src/core/ext/filters/backend_metrics/backend_metric_filter.h

```diff
@@ -31,20 +31,34 @@
 
 namespace grpc_core {
 
-class BackendMetricFilter : public ChannelFilter {
+class LegacyBackendMetricFilter : public ChannelFilter {
  public:
   static const grpc_channel_filter kFilter;
 
-  static absl::StatusOr<BackendMetricFilter> Create(const ChannelArgs& args,
-                                                    ChannelFilter::Args);
+  static absl::StatusOr<LegacyBackendMetricFilter> Create(
+      const ChannelArgs& args, ChannelFilter::Args);
 
   // Construct a promise for one call.
   ArenaPromise<ServerMetadataHandle> MakeCallPromise(
       CallArgs call_args, NextPromiseFactory next_promise_factory) override;
+};
+
+class BackendMetricFilter : public ImplementChannelFilter<BackendMetricFilter> {
+ public:
+  static const grpc_channel_filter kFilter;
+
+  static absl::StatusOr<BackendMetricFilter> Create(const ChannelArgs& args,
```

Comment:
I don't see this method defined anywhere?

Reply:
likely explains why none of the tests compile (will fix :))